packages/x-charts/src/ChartsReferenceLine/ChartsYReferenceLine.tsx

```diff
@@ -106,8 +111,15 @@ function ChartsYReferenceLine(props: ChartsYReferenceLineProps) {
 
   const classes = getYReferenceLineClasses(inClasses);
 
-  const spacingX = typeof spacing === 'object' ? (spacing.x ?? 0) : spacing;
-  const spacingY = typeof spacing === 'object' ? (spacing.y ?? 0) : spacing;
+  const defaultSpacingOtherAxis =
+    labelAlign === 'middle' || labelAlign === undefined
+      ? DEFAULT_SPACING_MIDDLE_OTHER_AXIS
+      : DEFAULT_SPACING;
+
+  const spacingX =
+    (typeof spacingProp === 'object' ? spacingProp.x : spacingProp) ?? defaultSpacingOtherAxis;
+  const spacingY =
+    (typeof spacingProp === 'object' ? spacingProp.y : spacingProp) ?? DEFAULT_SPACING;
```

Comment:
What about moving all this logic inside the `getTextParams()`?


```diff
const textParams = {
-    y: yPosition - spacingY,
     text: label,
     fontSize: 12,
     ...getTextParams({
       left,
       width,
-      spacingX,
       labelAlign,
+      spacingProp,
     }),
     className: classes.label,
   };
```

This would giv you a single function to manage x/y coordinates, so easier to test with unit tests